src/main/java/com/gluonhq/RunMojo.java

```diff
@@ -112,7 +120,11 @@ public void execute() throws MojoExecutionException {
         }
 
         invocationRequest.setPomFile(runPomFile);
-        invocationRequest.setGoals(Collections.singletonList("javafx:run"));
+        if (execution == null) {
+            invocationRequest.setGoals(Collections.singletonList("javafx:run"));
+        } else {
+            invocationRequest.setGoals(Collections.singletonList("javafx:run@" + execution.getExecutionId()));
+        }
```

Comment:
Below these lines there is an exception that prints `"javafx:run"`, probably better to print the final goal.
I guess you can refactor that to something like:
```
String goal = "javafx:run";
        if (execution != null) {
            goal += "@" + execution.getExecutionId();
        }
        invocationRequest.setGoals(Collections.singletonList(goal));
```